### ISSUE TITLE
Need complex number tests for log, log2, and log10

### ISSUE THREAD

**Issue by tbrowder** [Raku/roast#862]
See remarks in roast file:

    S32-num/log.t

A mathematician (such as @antononcube) needs to review adequacy of existing tests).

**Comment by librasteve** [comment]
```
use v6;
use Test;

# Tests for log, log2, and log10 of complex numbers

# Complex numbers to test
my $z1 = 1+1i;
my $z2 = -1+1i;
my $z3 = -2-3i;

# Expected values calculated manually or from reliable references
my $log_z1  = log($z1);  # Natural log of complex number
my $log_z2  = log($z2);
my $log_z3  = log($z3);

my $log2_z1 = log($z1) / log(2);   # Log base 2
my $log2_z2 = log($z2) / log(2);
my $log2_z3 = log($z3) / log(2);

my $log10_z1 = log($z1) / log(10); # Log base 10
my $log10_z2 = log($z2) / log(10);
my $log10_z3 = log($z3) / log(10);

# Start writing tests
plan 9;

# log (natural logarithm)
is log($z1), $log_z1, "log(1+1i) matches expected value";
is log($z2), $log_z2, "log(-1+1i) matches expected value";
is log($z3), $log_z3, "log(-2-3i) matches expected value";

# log2 (base 2 logarithm)
is log2($z1), $log2_z1, "log2(1+1i) matches expected value";
is log2($z2), $log2_z2, "log2(-1+1i) matches expected value";
is log2($z3), $log2_z3, "log2(-2-3i) matches expected value";

# log10 (base 10 logarithm)
is log10($z1), $log10_z1, "log10(1+1i) matches expected value";
is log10($z2), $log10_z2, "log10(-1+1i) matches expected value";
is log10($z3), $log10_z3, "log10(-2-3i) matches expected value";
```

**Comment by librasteve** [comment]
This should do the trick:

```
use v6; 
use Test;

# Tests for log, log2, and log10 of complex numbers

# Complex numbers to test
my $z1 = 1+i;
my $z2 = -1+i;
my $z3 = -2-3i;

# Expected values calculated manually or from reliable references

# Natural log 
my $log_z1  = 0.34657359 + 0.785398163i;
my $log_z2  = 0.34657359 + 2.35619449i;
my $log_z3  = 1.28247468 - 2.15879893i;

# Log base 2
my $log2_z1 = log($z1) / log(2);   
my $log2_z2 = log($z2) / log(2);
my $log2_z3 = log($z3) / log(2);

# Log base 10
my $log10_z1 = log($z1) / log(10); 
my $log10_z2 = log($z2) / log(10);
my $log10_z3 = log($z3) / log(10);

# Start writing tests
plan 9;

# log (natural logarithm)
is-approx log($z1), $log_z1, "log(1+1i) approximately matches expected value";
is-approx log($z2), $log_z2, "log(-1+1i) approximately matches expected value";
is-approx log($z3), $log_z3, "log(-2-3i) approximately matches expected value";

# log2 (base 2 logarithm)
is-approx log2($z1), $log2_z1, "log2(1+1i) approximately matches expected value";
is-approx log2($z2), $log2_z2, "log2(-1+1i) approximately matches expected value";
is-approx log2($z3), $log2_z3, "log2(-2-3i) approximately matches expected value";

# log10 (base 10 logarithm)
is-approx log10($z1), $log10_z1, "log10(1+1i) approximately matches expected value";
is-approx log10($z2), $log10_z2, "log10(-1+1i) approximately matches expected value";
is-approx log10($z3), $log10_z3, "log10(-2-3i) approximately matches expected value";
```

**Comment by librasteve** [comment]
btw I have deliberately switched to bare i literal since imo that is the most convenient notation and it is supported with the current implementation

**Comment by tbrowder** [comment]
That looks like a good suite of tests to round out the current few. I will add some Wikipedia links to some decent looking math references.

**Comment by tbrowder** [comment]
Closed with merge of PR #865
